### ISSUE TITLE
Delete `new-chat-mention` and `entity-event` notification categories

### ISSUE THREAD

**Issue by timolegros** [hicommonwealth/commonwealth#4370]
## Description
Chat was deleted in #4104 so the `new-chat-mention` `NotificationCategory` and the associated subscriptions (and notifications/notifications read) are no longer needed. There are currently 108,662 `new-chat-mention` subscriptions that can be removed.

Since CET removal we don't support event or entity-related subscriptions so `entity-event` should be removed from `NotificationCategories`.

## Engineering Requirements

- Removed `new-chat-mention` subscriptions, notifications, and notifications read as well as the category itself.
- Removed `entity-event` from the available notification categories.

**Comment by timolegros** [comment]
This issue should be merged and deployed after #4369 has been deployed for 24 hours. This is because #4369 greatly reduces the size of the Subscriptions, Notifications, and NotificationsRead tables leading to much better query performance. The 24-hour wait is because the database cleaner (which reduces table size) will only run at 2 AM UTC. It also gives time for the database to vacuum the dead rows thus further improving performance. This merge order aligns with the order stated in #4373.

**Comment by CowMuon** [comment]
PR is still in draft, pls re-submit when ready.

**Comment by timolegros** [comment]
@CowMuon please note that both issues you moved back _are_ ready. They are in draft to prevent merging in the incorrect order as described on this issue. https://github.com/hicommonwealth/commonwealth/pull/4479#issuecomment-1633109452

**Comment by timolegros** [comment]
Was ready but briefly putting back in progress to remove additional NotificationCategories that I discovered are not used while working on #4366.

**Comment by timolegros** [comment]
Please note this should not be merged until #4594 is merged _and deployed_ to production.

**Comment by timolegros** [comment]
This can be merged, but it should not be deployed until the database cleaner, which was deployed in #4594, is executed. The reason for this is that the database cleaner greatly reduces the size of the Notifications and Subscriptions tables, thus enhancing the speed of the migration in this PR, as well as in subsequent PRs.

**Comment by timolegros** [comment]
Ready for merging and deployment now that the database cleaner has run.
